lib/driver.rb

```diff
@@ -0,0 +1,67 @@
+require 'csv'
+require_relative 'csv_record'
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :name, :vin, :trips
+    attr_accessor :status
+    
+    def initialize(id: , name: , vin: , status: :AVAILABLE, trips: nil)
+      super(id)
+      @name = name
+      @vin = vin
+      @status = status
+      @trips = trips || []
+      
+      if vin.length !=17
+        raise ArgumentError, "Vin number must be string of length 17"
+      end
+      
+      if !([:AVAILABLE, :UNAVAILABLE].any? { |x| x == status })
+        raise ArgumentError, "Status is #{status}; needs to be :AVAILABLE or :UNAVAILABLE"
+      end
+    end
+
+    def add_trip(trip)
+      @trips << trip
+    end
+
+    def average_rating
+      total_rating = 0
+      number_of_finished_trips = 0
+      @trips.each do |trip|
+        if trip.end_time != nil
+          total_rating += trip.rating
+          number_of_finished_trips += 1
+        end
```

Comment:
Just noting you can simplify this a bit with enumerables

```suggestion
      return 0 if trips.length == 0

      completed_trips = trips.select { |trip| !trip.rating.nil? }
      total_ratings = completed_trips.reduce(0) do |sum, trip | 
        sum + trip.rating 
      end

      return total_ratings / completed_trips.length.to_f
```